unison-cli/src/Unison/CommandLine/Completion.hs

```diff
@@ -196,12 +196,6 @@ completeWithinNamespace compTypes query currentPath = do
     namesInBranch :: Int -> V2Branch.Branch Sqlite.Transaction -> Sqlite.Transaction [(CompletionType, Bool, Text)]
     namesInBranch hashLen b = do
       nonEmptyChildren <- V2Branch.nonEmptyChildren b
-      let textifyHQ :: (NameSegment -> r -> HQ'.HashQualified NameSegment) -> Map NameSegment (Map r metadata) -> [(Bool, Text)]
```

Comment:
this one was weird... not sure what happens.  

Without it we get this really weird error where it's expecting the wrong type!  Curious if anybody has any idea what's happening :thinking: 

```
src/Unison/CommandLine/Completion.hs:213:68: error: [GHC-83865]
    • Couldn't match type: Reference.Reference' Text Unison.Hash.Hash
                     with: Referent.Referent' Reference.Reference Reference.Reference
      Expected: NameSegment
                -> Referent.Referent -> HQ'.HashQualified NameSegment
        Actual: NameSegment
                -> Reference.Reference -> HQ'.HashQualified NameSegment
      Type synonyms expanded:
      Expected type: NameSegment
                     -> Referent.Referent' Reference.Reference Reference.Reference
                     -> HQ'.HashQualified NameSegment
        Actual type: NameSegment
                     -> Reference.Reference' Text Unison.Hash.Hash
                     -> HQ'.HashQualified NameSegment
    • In the first argument of ‘textifyHQ’, namely
        ‘(hqFromNamedV2Reference hashLen)’
      In the first argument of ‘($)’, namely
        ‘textifyHQ (hqFromNamedV2Reference hashLen)’
      In the second argument of ‘map’, namely
        ‘(textifyHQ (hqFromNamedV2Reference hashLen) $ V2Branch.types b)’
    |
213 |               (map (\(x, y) -> (TypeCompletion, x, y)) (textifyHQ (hqFromNamedV2Reference hashLen) $ V2Branch.types b)),                                                                                       b
    |                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.stack-work/install/x86_64-linux/a063c0a73e60c13f68571667e6977d457cc001cca1e544d0d7876989a155d05e/9.2.8/lib
src/Unison/CommandLine/Completion.hs:213:102: error: [GHC-83865]
    • Couldn't match type: Reference.Reference' Text Unison.Hash.Hash
                     with: Referent.Referent' Reference.Reference Reference.Reference
      Expected: Map
                  NameSegment
                  (Map Referent.Referent (Sqlite.Transaction V2Branch.MdValues))
        Actual: Map
                  NameSegment
                  (Map Reference.Reference (Sqlite.Transaction V2Branch.MdValues))
      Type synonyms expanded:
      Expected type: Map
                       NameSegment
                       (Map
                          (Referent.Referent' Reference.Reference Reference.Reference)
                          (Sqlite.Transaction V2Branch.MdValues))
        Actual type: Map
                       NameSegment
                       (Map
                          (Reference.Reference' Text Unison.Hash.Hash)
                          (Sqlite.Transaction V2Branch.MdValues))
    • In the second argument of ‘($)’, namely ‘V2Branch.types b’
      In the second argument of ‘map’, namely
        ‘(textifyHQ (hqFromNamedV2Reference hashLen) $ V2Branch.types b)’
      In the second argument of ‘Monoid.whenM’, namely
        ‘(map
            (\ (x, y) -> (TypeCompletion, x, y))
            (textifyHQ (hqFromNamedV2Reference hashLen) $ V2Branch.types b))’
    |
213 |               (map (\(x, y) -> (TypeCompletion, x, y)) (textifyHQ (hqFromNamedV2Reference hashLen) $ V2Branch.types b)),
    |                                                                                                      ^^^^^^^^^^^^^^^^
```

Reply:
Yes, without the `forall r.`, GHC infers local functions to have a concrete type according to the first encountered usage. This one must be getting used more than once with different `r`.  Why remove the type annotation though?

Reply:
Yeah so that was my thinking as well initially so I tried to add a `forall r.` in the type definition but it was still erroring out :shrug: 

> Why remove the type annotation though?

Ah sorry, I should've selected the whole change when adding my comment: I just moved the whole function so here it only shows up as the removed line but if you look at the diff I re-added it a few lines below (not sure if that's what you meant?)